pkg/sql/execinfra/processorsbase.go

```diff
@@ -840,7 +840,7 @@ func ProcessorSpan(
 	eventListeners ...tracing.EventListener,
 ) (context.Context, *tracing.Span) {
 	sp := tracing.SpanFromContext(ctx)
-	if sp == nil {
+	if sp == nil || sp.IsNoop() {
```

Comment:
This is quite a subtle catch. Do you think we should unify `nil` and `IsNoop()`? It seems like a footgun that you have to remember to check them together.

Reply:
Yeah, this ties in well with my other comment above[^1] - I'm unclear on what the intention of all of this code was and assume what we _want_ has drifted now, too.
We sort-of-mostly moved into a world where there are trace spans everywhere. I think the idea was that you have a noopSpan unless tracing is off? Today I think that's a bad idea too though, since you end up allocating a bunch just to wrap `context.Context`. I don't believe that any version of "always-on tracing" is going to be the future and even if it were, we've been paying all of the cost and getting little benefit. So we'd likely want to figure out what it is we want (for example: no spans by default) and clean up with that goal. That isn't something that I can clean up as a drive-by though, I can only offer the lame suggestion to file a follow-up issue against Obs.

[^1]: https://github.com/cockroachdb/cockroach/pull/135146#discussion_r1847829820